.github/workflows/codeql-analysis.yml

```diff
@@ -25,15 +25,36 @@ jobs:
     - name: Checkout repository
       uses: actions/checkout@v4
 
+    - uses: actions/setup-java@v4
+      with:
+        java-version: 17
+        distribution: temurin
+
+    - name: Cache local Maven repository
+      uses: actions/cache@v3
+      with:
+        path: |
+          ~/.m2/repository/io/opentelemetry/
+        key: ${{ runner.os }}-maven-local-${{ hashFiles('.github/patches/opentelemetry-java*.patch') }}
+
+    - name: Publish patched dependencies to maven local
+      uses: ./.github/actions/patch-dependencies
```

Comment:
@bryan-aguilar should we have a if condition here, like below?
`        if: steps.cache-local-maven-repo.outputs.cache-hit != 'true'
`

Reply:
added